ili9341/spi_stm32f4.go

```diff
@@ -0,0 +1,142 @@
+// +build stm32f4
+
+package ili9341
+
+import (
+	"device/stm32"
+	"machine"
+)
+
+type spiDriver struct {
+	bus machine.SPI
+}
+
+func NewSPI(bus machine.SPI, dc, cs, rst machine.Pin) *Device {
+	return &Device{
+		dc:  dc,
+		cs:  cs,
+		rst: rst,
+		rd:  machine.NoPin,
+		driver: &spiDriver{
+			bus: bus,
+		},
+	}
+}
+
+func (pd *spiDriver) configure(config *Config) {
+}
+
+func (pd *spiDriver) write8(b byte) {
+	if !pd.bus.Bus.CR1.HasBits(stm32.SPI_CR1_SPE) {
+		pd.bus.Bus.CR1.SetBits(stm32.SPI_CR1_SPE)
+	}
+
+	pd.setWord(b, true, true)
+
+	pd.bus.Bus.CR1.ClearBits(stm32.SPI_CR1_SPE)
+}
+
```

Comment:
The processing for `SPI_CR1_SPE` does not seem to be necessary, and removing it will increase the speed from 68 fps to 88 fps.
Other functions are the same.

Any reason to use `SPI_CR1_SPE` ?

```suggestion
func (pd *spiDriver) write8(b byte) {
	pd.setWord(b, true, true)
}
```

Reply:
I wasn't sure about that either, perhaps it clears certain flags (e.g., overrun/underrun?) in the SPI peripheral once the transaction has completed? You wouldn't want them to persist across transactions. 

But for good connections with a well-behaving slave, you can probably get by **without** doing it. 

Note that the motivation for this disabling and reenabling the peripheral comes from STM32CubeIDE-generated driver code, which also performs this set/clear on CR1 SPE in its Transmit/Receive functions - both their HAL driver and their LL driver do this.

For reference, I will attach my STM32CubeIDE project configured for the STM32F405 Feather. It has UART, I2C, and SPI interfaces all configured on their correct pins, and the system clock is configured with the 12 MHz HSE crystal onboard the feather.

Reply:
I suggest removing this logic from the ILI9341 driver as you're implying. 

This logic would be better suited in `(SPI).Transfer` where robustness and bus stability is more important than performance.

Do you agree? Or would you remove it altogether?

Reply:
In my opinion, it should be removed from ILI9341.
And as you say, I think we should leave it in SPI.Transfer().


https://github.com/tinygo-org/tinygo/blob/7fc5095b6b768a6e441752223724b27adaf2f014/src/machine/machine_stm32_spi.go#L150